ipn/ipnlocal/cert.go

```diff
@@ -356,9 +370,25 @@ type certStateStore struct {
 	// This field allows a test to override the CA root(s) for certificate
 	// verification. If nil the default system pool is used.
 	testRoots *x509.CertPool
+	readOnly  bool
 }
 
 func (s certStateStore) Read(domain string, now time.Time) (*TLSCertKeyPair, error) {
+	// If we're using a store that supports atomic reads, use that
+	if kr, ok := s.StateStore.(interface {
+		ReadTLSCertAndKey(string) ([]byte, []byte, error)
```

Comment:
This is not implemented by any store yet- but will be implemented by kube store in HA mode, so that it can perform custom logic when retrieving certs and keys, such as attempt to read from a named Secret if not found in memory store